marimbabot_speech/src/speech_synthesis.py

```diff
@@ -13,22 +14,33 @@ def __init__(self):
 
         # listens to the defined text responses from behavior node
         self.response_sub = rospy.Subscriber("behavior_node/response", String, self.callback_response)
+        self.audio_pub = rospy.Publisher('robotsound', SoundRequest, queue_size=1)
 
     def callback_response(self, response):
         text = response.data
 
-        mimic_subprocess = subprocess.Popen(
-            ('mimic3', '--voice', self.voice, '--length-scale', self.speed, text), 
-            stdout=subprocess.PIPE)
-
-        # Play the audio from the previous process with aplay
-        aplay_subprocess = subprocess.Popen(
-            ('aplay', '-'), 
-            stdin=mimic_subprocess.stdout, 
-            stdout=subprocess.PIPE)
-        
-        # Wait for the process to finish
-        aplay_subprocess.wait()
+        temp_ = tempfile.TemporaryFile()
```

Comment:
You could start a background thread that deletes the temp files after let's say 5 minutes. But I think storage is cheap and the files are tiny so it shouldn't matter.

Reply:
I replaced the tempfile creation with the built-in sound_play TTS as a solution for the tempfile problem. It does not sound as pretty but the response is much faster. I left the old code commented out in case we want to switch back.